lib/matplotlib/legend.py

```diff
@@ -1105,6 +1105,11 @@ def _find_best_position(self, width, height, renderer, consider=None):
         assert self.isaxes
 
         verts, bboxes, lines, offsets = self._auto_legend_data()
+        if len(verts) + len(bboxes) + len(lines) + len(offsets) > 10000:
```

Comment:
Should this be taken from a new rcParam? Should it be increased? Decreased?

Reply:
Concerning the value: Can you time it? I’d say the maximum runtime for legend placement without a warning should be 0.5-1s. I think this can be hard-coded. Users can filter the warning.

Reply:
Increased to 500,000 which was the point at which interaction became painful for me on a laptop.